pkg/db/db_session/default.go

```diff
@@ -112,20 +153,33 @@ func waitForNotification(ctx context.Context, l *pq.Listener, callback func(id s
 		case <-time.After(10 * time.Second):
 			logger.V(10).Infof("Received no events on channel during interval. Pinging source")
 			go func() {
+				// TODO: Need to handle the error, especially in cases of network failure.
 				l.Ping()
 			}()
 		}
 	}
 }
 
-func newListener(ctx context.Context, connstr, channel string, callback func(id string)) {
+func newListener(ctx context.Context, dbConfig *config.DatabaseConfig, channel string, callback func(id string)) {
 	logger := ocmlogger.NewOCMLogger(ctx)
 
 	plog := func(ev pq.ListenerEventType, err error) {
 		if err != nil {
 			logger.Error(err.Error())
 		}
 	}
+	connstr := dbConfig.ConnectionString(true)
+	// append the password to the connection string
+	if dbConfig.AuthMethod == constants.AuthMethodPassword {
+		connstr += fmt.Sprintf(" password='%s'", dbConfig.Password)
+	} else if dbConfig.AuthMethod == constants.AuthMethodMicrosoftEntra {
+		token, err := getAccessToken(ctx, dbConfig)
+		if err != nil {
+			panic(err)
+		}
+		connstr += fmt.Sprintf(" password='%s'", token.Token)
```

Comment:
What's the behaviour of this one when `sslrootcert` is set to an empty string?

Reply:
I've noticed weird behaviour related to this e.g
```
2024-12-17T07:17:16.531Z	ERROR	logger/logger.go:129	  pq: empty password returned by client 
github.com/openshift-online/maestro/pkg/logger.(*logger).log
	/maestro/pkg/logger/logger.go:129
github.com/openshift-online/maestro/pkg/logger.(*logger).Error
	/maestro/pkg/logger/logger.go:120
github.com/openshift-online/maestro/pkg/db/db_session.newListener.func1
	/maestro/pkg/db/db_session/default.go:168
github.com/lib/pq.(*Listener).emitEvent
	/go/pkg/mod/github.com/lib/pq@v1.10.7/notify.go:797
github.com/lib/pq.(*Listener).listenerConnLoop
	/go/pkg/mod/github.com/lib/pq@v1.10.7/notify.go:817
github.com/lib/pq.(*Listener).listenerMain
	/go/pkg/mod/github.com/lib/pq@v1.10.7/notify.go:856
```	

And I think we'll need to quote the sslrootcert when creating the connection string, similar as to what we do with the password here.